flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -457,8 +486,9 @@ def rx_weight_fn(code):
     """A function for returning the weight from the common vertex."""
```

Comment:
Why is `code` the argument of this function? Should this function be instead:

```
    def rx_weight_fn(edge):
        if edge["common_vertex"] is not None:
            return float(code.graph.nodes[edge["common_vertex"]]["weight"])
        weight = edge.get("weight")
        if weight is not None:
            return float(weight)
        return 0.0
```
and then above you would set weight_fn=rx_weight_fn ?

Reply:
It is a function taking a code and returning a function from edge to weight. 

For the version you suggest, how do you access code? You use it without providing it. 

Technically, this is what we call a closure in programming jargon.

See for more information 

https://stackoverflow.com/questions/36636/what-is-a-closure#7464475

Reply:
Sorry - my suggestion was for the previous version of the function. But the updated version does not refer to code anymore, right? So would this work now?

```
    def rx_weight_fn(edge):
        weight = edge.get("weight")
        if weight is not None:
            return float(weight)
        return 0.0
```